warehouse/bigquery.go

```diff
@@ -163,13 +168,92 @@ func (bq *BigQuery) LoadToWarehouse(filename string, bundles ...fullstory.Export
 	// start and wait on loading job
 	job, err := loader.Run(bq.ctx)
 	if err != nil {
-		log.Printf("Could not start BQ load job for file %s", filename)
+		log.Printf("Could not start BQ load job for file %s", objName)
 		return err
 	}
 
 	return bq.waitForJob(job)
 }
 
+// EnsureCompatibleExportTable ensures that the all the fields present in the hauser schema are present in the BigQuery table schema
+// If the table exists, it compares the schema of the export table in BigQuery to the schema in hauser and adds any missing fields
+// if the table doesn't exist, it creates a new export table with all the fields specified in hauser
+func (bq *BigQuery) EnsureCompatibleExportTable() error {
+	// Get Hauser Schema
+	// this is required if we create a new table or if we have to compare to the existing table schema
+	hauserSchema, err := bigquery.InferSchema(bundleEvent{})
+	if err != nil {
+		return err
+	}
+
+	if err := bq.connectToBQ(); err != nil {
+		log.Fatal(err)
+	}
+	defer bq.bqClient.Close()
+
+	if bq.doesTableExist(bq.conf.BigQuery.ExportTable) {
+		log.Printf("Export table exists, making sure the schema in BigQuery is compatible with the schema specified in Hauser")
+
+		// get current table schema in BigQuery
+		table := bq.bqClient.Dataset(bq.conf.BigQuery.Dataset).Table(bq.conf.BigQuery.ExportTable)
+		md, err := table.Metadata(context.Background())
+		if err != nil {
+			return err
+		}
+
+		// Find the fields from the hauser schema that are missing from the BiqQuery table
+		missingFields := bq.GetMissingFields(hauserSchema, md.Schema)
+
+		// If fields are missing, we add them to the table schema
+		if len(missingFields) > 0 {
+			// Append missing fields to export table schema
+			md.Schema = append(md.Schema, missingFields...)
+			if err := bq.updateTable(table, md.Schema); err != nil {
+				return nil
+			}
+		}
+	} else {
```

Comment:
style nit: Following up with jess's comment on early returns, you could probably check for this condition first and reduce one layer of nesting.

```go
if bq.doesTableExist(bq.conf.BigQuery.ExportTable) {
  // Table does not exist, create new table
  log.Printf("Export table does not exist, creating one.")
  if err := bq.createExportTable(hauserSchema); err != nil {
     return err
  }
  return nil
}

// carry out actions on table that exists.
```

Reply:
ditto for method in redshift.